js/script.js

```diff
@@ -1,120 +1,164 @@
 import {
-  $taskInput,
-  $addButton,
-  $taskList,
+  $mainContainer,
   $searchInput,
   $searchButton,
+  $splash,
+  $noTask,
+  $taskListContainer,
+  $loadMore,
+  $createButton,
 } from "./elements.js";
 import {
   showToastMessage,
-  createTaskElement,
   sanitizeInput,
+  toggleInputContainer,
+  containerBuilder,
+  showSpinnerOverlay,
+  hideSpinnerOverlay,
   formatDate,
 } from "./utilities.js";
+import { MESSAGES } from "./const.js";
 
 let tasks = [];
+let isVisible = false;
+
+const addButtonHandler = (container) => {
+  isVisible = !isVisible;
+  const taskTitle = sanitizeInput(document.getElementById("task-input").value);
 
-const addButtonHandler = () => {
-  const taskTitle = sanitizeInput($taskInput.value);
   if (taskTitle) {
-    createTask(taskTitle);
-    showToastMessage("Task added successfully!");
-    $addButton.disabled = true;
+    const $overlay = showSpinnerOverlay(container);
+    setTimeout(() => {
+      createTask(taskTitle);
+      showToastMessage(MESSAGES.SUCCESS, true);
+      hideSpinnerOverlay($overlay);
+    }, 1000);
```

Comment:
these code block is repetitive in several places, could you please make it as a reusable function and use it?